src/lib/fixed_string.hpp

```diff
@@ -0,0 +1,117 @@
+                "Other FixedString is longer than current maximum string length");
+    const auto copied_length = other.maximum_length() < _maximum_length ? other.maximum_length() : _maximum_length;
+    other._copy_to(_mem, copied_length);
+    // Fill unused fields of char array with null terminator, in order to overwrite the content of
+    // the old FixedString. This is especially important if the old FixedString was longer than the other FixedString.
+    if (copied_length < _maximum_length) {
+      memset(_mem + copied_length, '\0', _maximum_length - copied_length);
+    }
+    return *this;
+  }
+
+  // Returns the length of the string
+  size_t size() const {
+    const auto position = std::find(_mem, _mem + _maximum_length, '\0');
+    return std::distance(_mem, position);
+  }
+
+  // Returns the maixmum possible size of storable strings
+  size_t maximum_length() const { return _maximum_length; }
+
+  // Creates a string object from FixedString
+  std::string string() const {
+    const auto string_value = std::string(_mem, _maximum_length);
+    const auto pos = string_value.find('\0');
+
+    if (pos == std::string::npos) {
+      return string_value;
+    } else {
+      return string_value.substr(0, pos);
+    }
+  }
+
+  // Compare FixedStrings by comparing the underlying char arrays.
+  // If one FixedString is longer than the other FixedString and the beginning of the longer FixedString
+  // is equal to the other FixedString, the shorter FixedString is smaller.
+  // Example: "defg" < "defghi"
+  bool operator<(const FixedString& other) const {
+    const auto smallest_length = size() < other.size() ? size() : other.size();
+    const auto result = memcmp(_mem, other._mem, smallest_length);
+    if (result == 0) return size() < other.size();
+    return result < 0;
+  }
+
+  // The FixedStrings must have the same length to be equal
+  bool operator==(const FixedString& other) const {
+    if (size() != other.size()) return false;
+    return memcmp(_mem, other._mem, size()) == 0;
+  }
+
+  // Prints FixedString as string
+  friend std::ostream& operator<<(std::ostream& os, const FixedString& obj) { return os << obj.string(); }
+
+  // Support swappable concept needed for sorting values. See: http://en.cppreference.com/w/cpp/concept/Swappable
+  friend void swap(FixedString& lhs, FixedString& rhs) { lhs.swap(rhs); }
+
+  // Swap two FixedStrings by exchanging the underlying memory's content
+  void swap(FixedString& other) {
+    DebugAssert(_maximum_length == other.maximum_length(),
+                "FixedStrings must have the same maximum_length in order to swap them");
+    std::swap_ranges(_mem, _mem + _maximum_length, other._mem);
+  }
+
+ protected:
+  char* const _mem;
+  const size_t _maximum_length;
+  const bool _owns_memory = true;
+
+  // Copy chars of current FixedString to a new destination
+  size_t _copy_to(char* s, size_t len, size_t pos = 0) const {
```

Comment:
This should have a `DebugAssert` making sure that the two memory areas don‘t overlap (UB)